Update CI Docker files

dgaliffiAMD opened the pull request:
## Motivation

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->
Installing json library in Dockerfiles for https://github.com/ROCm/rocm-systems/pull/1033.
Installing other packages what will be used soon.
Fixing CI container update workflow.

## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->
Add `nlohmann-json-dev` (or equivalent) to CI Docker images for RHEL, SUSE, and Ubuntu.
Add `gmock-dev` and `gtest-dev` (or equivalent) to CI Docker images for RHEL, SUSE, and Ubuntu.
Add `--set solver classic` to conda config to resolve an issue setting up the conda environment

## Test Plan

<!-- Explain any relevant testing done to verify this PR. -->
Running update Docker workflow to update all of our CI dockers and perform a test build with each.

## Test Result

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

dgaliffiAMD committed:
Installing `nlohmann_json-devel` on CI dockers

mradosav-amd commented:
Maybe we can also include `libgtest-dev` and `libgmock-dev` packages in this PR. It would be easier in future when we add google tests.

Ubuntu/Debian: `libgtest-dev` and `libgmock-dev`
RHEL: `gtest-devel` and `gmock-devel`
OpenSUSE: `gtest-devel` and `gmock-devel`

dgaliffiAMD committed:
Add `--set solver classic` to conda config

Fixes error during conda installation.

```
CondaValueError: You have chosen a non-default solver backend (libmamba) but it was not recognized. Choose one of: classic
```

dgaliffiAMD committed:
Adding `gmock` and `gtest` packages

dgaliffiAMD committed:
Fix the `gmock` and `gtest` package names for opensuse

dgaliffiAMD committed:
Updated OSes for CI Containers

jbonnell-amd committed:
Update logic for running with --break-system-packages, add perfetto install check

dgaliffiAMD committed:
Update Dockerfile.ubuntu.ci

Updated OS_VERSION check used in pip install
